dev-manuals/services/integrate-new-minigame.md

```diff
@@ -0,0 +1,339 @@
+  questionText: string;
+  answer: string;
+
+  public constructor(questionText: string, answer: string) {
+    this.questionText = questionText;
+    this.answer = answer;
+  }
+}
+```
+
+### Add a REST client
+
+Add the URL of the backend in the `config.ts`
+```typescript
+<YOUR_MINIGAME_NAME>ApiUrl: "/minigames/<YOUR_MINIGAME_NAME>/api/v1",
+```
+
+Add a class <YOUR_MINIGAME_NAME>-rest-client.ts in package `ts.rest-clients`, e.g.
+```typescript
+import axios, { AxiosResponse } from "axios";
+
+import config from "@/config";
+
+import { <YOUR_MINIGAME_NAME>Configuration } from "@/ts/models/<YOUR_MINIGAME_NAME>-models";
+
+export async function post<YOUR_MINIGAME_NAME>Config(
+  <YOUR_MINIGAME_NAME>Config: <YOUR_MINIGAME_NAME>Configuration
+): Promise<AxiosResponse> {
+  return axios.post(
+    `${config.<YOUR_MINIGAME_NAME>ApiUrl}/configurations`,
+    <YOUR_MINIGAME_NAME>Config
+  );
+}
+
+export async function get<YOUR_MINIGAME_NAME>Config(
+  id: string
+): Promise<AxiosResponse> {
+  return axios.get(`${config.<YOUR_MINIGAME_NAME>ApiUrl}/configurations/${id}`);
+}
+```
+
+### Create a configuration modal
+
+Create a new configuration modal under the package `components.EditMinigameModals`.  
+We can't document exactly what the code should look like, as it varies a lot, but you can orient yourself on the existing edit components.  
+It is important that the minigame configurations are loaded and updated in the component.  
+When a modal gets closed, emit `closedModal` and when a configuration gets updated emit `updateMinigameConfiguration`.
+
+The minigames edit view is in the class `MinigameTasksView.vue` under the package `views`.
+
+Create a new variable:
+```typescript
+const show<YOUR_MINIGAME_NAME>Modal = ref(false);
+```
+
+Add your minigame in the `editMinigameConfiguration` method:
+```typescript
+case Minigame.<YOUR_MINIGAME_NAME>:
+    show<YOUR_MINIGAME_NAME>Modal.value = true;
+    break;
+```
+
+Append the following at the bottom of the `closedEditModal` method:
+```typescript
+show<YOUR_MINIGAME_NAME>Modal.value = false;
+```
+
+Add the new edit modal. Put the modal after the other modals at the bottom of the file.
+```typescript
+<Edit<YOUR_MINIGAME_NAME>Modal
+    :showModal="show<YOUR_MINIGAME_NAME>Modal"
+    :minigame="editedMinigame"
+    @updateMinigameConfiguration="updateMinigameConfiguration"
+    @closedModal="closedEditModal"
+  />
+```
+
+#### Add configuration import/ export
+
+Go in the edit modal component of the minigame. 
+You need following imports:
+```typescript
+import { saveAs } from "file-saver";
+import { arrayOf, defaultValue, object, string, int } from "checkeasy"; // depends on your configuration attributes
+import { importConfiguration } from "@/ts/import-configuration";
+import ImportExportConfiguration from "@/components/ImportExportConfiguration.vue";
+```
+
+To make a configuration downloadable, create the method downloadConfiguration:
+```typescript
+function downloadConfiguration() {
```

Comment:
```suggestion
To make a configuration downloadable, create the method `downloadConfiguration`:
```typescript
// configuration is expected to be a ref to the current configuration
function downloadConfiguration() {
```

Reply:
That was a bit too fast. There is still an open suggestion.